models/app.models.js

```diff
@@ -47,44 +46,61 @@ exports.selectArticles = () => {
   });
 };
 
-
 exports.selectComments = (article_id) => {
   let queryString = `SELECT * FROM comments`;
   const queryValue = [];
-  const queryProms = []
+  const queryProms = [];
   if (article_id) {
     queryString += ` WHERE article_id = $1`;
     queryValue.push(article_id);
-    queryProms.push(checkExists("articles", "article_id", article_id))
+    queryProms.push(checkExists("articles", "article_id", article_id));
   }
   queryString += ` ORDER BY created_at DESC`;
-  queryProms.push(db.query(queryString, queryValue))
-  return Promise.all(queryProms).then((promResults) =>{
-    if (queryProms.length === 1){
-      return promResults[0].rows
+  queryProms.push(db.query(queryString, queryValue));
+  return Promise.all(queryProms).then((promResults) => {
+    if (queryProms.length === 1) {
+      return promResults[0].rows;
     } else {
-      return promResults[1].rows
+      return promResults[1].rows;
     }
-  })
+  });
 };
 
 exports.postNewComment = (article_id, commentToPost) => {
   const { body, author } = commentToPost;
   let queryString = `INSERT INTO comments (body, article_id, author) 
                     VALUES ($1, $2, $3) RETURNING author as username, body`;
-  const queryValue = []
+  const queryValue = [];
+  const queryProms = [];
+  if (article_id && commentToPost) {
+    queryValue.push(body, article_id, author);
+    queryProms.push(checkExists("articles", "article_id", article_id));
+  }
```

Comment:
Article_id is a foreign key, so you don't need to do a manual check that it exists. There will also always be an article_id because you are only getting to this point if your endpoint is /api/articles/:somearticleid <- whatever that id is